### PR TITLE
New relic and artillery

### PR DESCRIPTION
**Pull request by masoncpott**
@sam-masno @sjamcox @tyler-thee-creator 

Added New Relic to proxy.
Confirmed New Relic is capturing data correctly.
Added Artillery to proxy.
Wrote Stress Test Config and Helper Functions.
Confirmed all tests work properly.

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -20,8 +24,49 @@ app.get('/priceandinventory/id/:productId', async (req, res) => {
...
     res.status(200).send(data);
   } catch (e) {
-    res.status(500).send(err);
+    res.status(500).send(e);
   }
-})
+});
+
+app.post('/priceandinventory/id/multiple', async (req, res) => {
+  const productIds = req.body
```
Comment (tyler-thee-creator) [inline server/index.js:32]: Nice looking routes, just a suggestion but you could destructure this and rename it while you do it too, so:

const productIds = req.body

would become:

const { body: productIds } = req;

totally not necessary and now that I write it out, it is a big longer than the original

File: server/index.js [hunk 2]
```diff
@@ -20,8 +24,49 @@ app.get('/priceandinventory/id/:productId', async (req, res) => {
...
+    const productInfo = await axios.post(`${ url }/priceandinventory/id/multiple`, productIds);
+    res.status(200).send(productInfo);
+  } catch (e) {
+    res.status(500).send(e);
+  }
+});
+
+app.post('/priceandinventory/id/createRecord', async (req, res) => {
+  const newRecord = req.body;
+  console.log('CREATE RECORD',req.body)
```
Comment (tyler-thee-creator) [inline server/index.js:43]: Might want to take this console log out later so when you're testing with artillery, it doesn't slow things down?

File: server/index.js [hunk 3]
```diff
@@ -20,8 +24,49 @@ app.get('/priceandinventory/id/:productId', async (req, res) => {
...
+    res.sendStatus(200);
+  } catch (e) {
+    res.status(500).send(e);
+  }
+});
+
+app.put('/priceandinventory/id/updateRecord', async (req, res) => {
+  const recordToUpdate = req.body;
+  try {
+    await axios.put(`${ url }/priceandinventory/id/updateRecord`, recordToUpdate);
```
Comment (tyler-thee-creator) [inline server/index.js:55]: Love that you're using the ${ url } with the axios call, clean and allows you to control your host in one place, nicely done

File: stressTest.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      name: "Ten users per second for 60 seconds"
+    - pause: 10
+    - duration: 60
+      arrivalRate: 100
+      name: "100 users per second for 60 seconds"
+    - pause: 10
+    - duration: 60
+      arrivalRate: 1000
+      name: "1000 users per second for 60 seconds"
+scenarios:
```
Comment (tyler-thee-creator) [inline stressTest.yml:20]: Nice job ramping up here, just a thought, would you also want to test GET and POST separately to see how each one performs on its own with increased RPS? Might not even be that important but was something I was curious about your opinion on

File: stressTestFunctions.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+const faker = require('faker');
+
+const randomNumber = (lowerLimit, upperLimit) => {
+  return faker.datatype.number({min: lowerLimit, max: upperLimit});
+};
+
+const generateOneRecord = () => {
```
Comment (tyler-thee-creator) [inline stressTestFunctions.js:7]: These artillery functions look awesome, nice job, like the randomNumber function which you use a bunch of times in this file, very clean

**Review by tyler-thee-creator — APPROVED** [review]
Looks solid, nice job

File: server/index.js (hunk 2 above)
Comment (masoncpott) [reply server/index.js:43]: good catch! I will remove that console log before I merge

File: server/index.js (hunk 3 above)
Comment (masoncpott) [reply server/index.js:55]: I think this will make things easier come deployment time

File: stressTest.yml (hunk 4 above)
Comment (masoncpott) [reply stressTest.yml:20]: Is testing the GET and POST separately how we should be conducting our tests? The instructions are clear on how exactly we should be conducting our tests, which is why I just went with one test file that does both.

**Pull request merged**